### PR TITLE
cps/adding 3d wake process

### PR DESCRIPTION
**Pull request by inigolcanalejo**
This PR is a follow up of https://github.com/KratosMultiphysics/Kratos/pull/5252 covering point 2 (add the first version of the define_wake_process_3d.py). I have a first proposal and I would like to discuss it especially with @marcnunezc and @rubenzorrilla.

This PR only adds the file define_wake_process_3d.py and its tests. Tests are added to check some corner cases (nodes laying on the wake and nodes from kutta elements above the wake but below the lower surface of the wing). There is also a list of TODOs (improvements) marked as comments that I will list also here below.

**The goal:** Implement a first version of the wake process that:

1. Selects elements cut by the wake as WAKE
2. Selects the KUTTA elements (touching the trailing edge from below and not cut)
3. Selects the wake_kutta elements (wake elements touching the trailing edge) (still marked as STRUCTURE)

The first version is thought to work the most simple case: "infinite wings" (wings going from side to side of the domain) with a straight trailing edge. This is considered to be a straight extension from the 2d case.

To show that it works, here is an example. The kutta_wake elements:
kutta_wake_elems

Another view:

kutta_wake_elems2

The kutta elements:

kutta_elements

All wake elements:

wake_elements

**The big question mark is whether it is ok to be using the CalculateDistanceToSkinProcess3D instead of the 
CalculateDiscontinuousDistanceToSkinProcess3D.** I refer here to https://github.com/KratosMultiphysics/Kratos/issues/5278

Note that this first version is in python cause implementation is faster but of course it will be eventually migrated once the algorithm of the process is consolidated.

List of TODOs and improvements in this process:

- Automatize the generation of the wake. Right now the wake is imported as an stl mesh. The goal is to automatically generate the wake from the trailing edge. Here I would like to thank @rubenzorrilla and @philbucher for their awesome ideas.
- Make it work for non straight trailing edges: here we need to check whether the current criteria to select the KUTTA elements still holds.
- Make it work with wings with finite span. Here there is some previous work required. We need to know what we want to do/try with the elements one the side of the wake, specially the ones touching the wing tips.
- Generalize for more than one lifting surface. (This comes later in the future but it might be a good idea to try this in 2D first with two airfoils and two wakes).
- Migrate to C++

### CODE REVIEW

File: applications/CompressiblePotentialFlowApplication/python_scripts/define_wake_process_3d.py [hunk 1]
```diff
@@ -0,0 +1,351 @@
...
+            err_msg = "Empty wake_stl_file_name in DefineWakeProcess3D\n"
+            err_msg += "Please specify the stl file name that contains the wake surface nodes"
+            raise Exception(err_msg)
+
+        self.epsilon = settings["epsilon"].GetDouble()
+        self.output_wake = settings["output_wake"].GetBool()
+
+        # For now plane wake surfaces are considered.
+        # TODO: Generalize this to curved wake surfaces
+        self.wake_normal = settings["wake_normal"].GetVector()
```
Comment (marcnunezc) [inline applications/CompressiblePotentialFlowApplication/python_scripts/define_wake_process_3d.py:61]: Is it possible to automate the computation of the wake_normal so it does not have to be defined in the json?

**Review by marcnunezc — APPROVED** [review]
Hi,

Nice work! I was waiting to add my review until the discussion in #5278 was finished. I am approving your pull request since you have defined all the corner cases and added tests to them, even though we use the continuous distance process to retrieve the elemental discontinuous distances. I think these tests can help to later validate possible changes to the discontinuous process to account for these kind of corner cases (i.e skin model part laying on a node, on a tetrahedron face, etc).

File: applications/CompressiblePotentialFlowApplication/python_scripts/define_wake_process_3d.py (hunk 1 above)
Comment (inigolcanalejo) [reply applications/CompressiblePotentialFlowApplication/python_scripts/define_wake_process_3d.py:61]: Good point. For a wing with a straight trailing edge we can compute the wake_normal using the wake direction and the span direction. I didn't implement this because I thought that anyway the goal is to generalize this for curved trailing edges.

For a wing with a curved trailing edge the wake normal will not be unique and we will need to compute the normal in each discretized triangle wake surface element and then pass this information to the cut fluid wake element to apply the wake condition.

Do you need this feature? If you want we can automatize it now for straight trailing edges and then see what we do for curved trailnig edges when the time comes.
